Allow to access the datatime object in epn

Et7f3 opened the pull request:
We have internationals users and we want to format in their native language.

Et7f3 committed:
ipa-epn: Provides expiration datetime object & days count in epn template

We have internationals users and we want to format in their native language.
Here we pass the raw datimeobject and user can now use strftime in jinja
template to format depending on user language.
This PR helps in situation where their is one instance of IPA by country
one node in country X can be configured with a date format X and node in
country Y can be confiugred with a date format Y.

Maybe a more general solution is to add a country attribute and make it
visibile to template so it can dynamically switch used template per user
in the single node case.

No related issue opened.

rcritten commented:
Please see https://www.freeipa.org/page/Contribute/Code on PR requirements.

The text of the commit doesn't match the content (this has nothing to do with i18n).

Will that strptime work? I think the value will be like 20240724200619Z

The time delta is going to display oddly, like: 3:59:59.999997

Et7f3 commented:
> The text of the commit doesn't match the content (this has nothing to do with i18n).

The title explain the modification: we provide the datetime object in variable for template. Thus in template we can do expiration_datetime.strfmt("with my native language").

> Will that strptime work ?
Yes it can parse date from FreeIPA server. I intended to let user format as their wish. I could Have added another setting with the python format.

> The time delta is going to display oddly, like: 3:59:59.999997

In template we can do .days or a formatting function.

Et7f3 commented:
> Please see https://www.freeipa.org/page/Contribute/Code on PR requirements.
> 
> The text of the commit doesn't match the content (this has nothing to do with i18n).

I have updated my (comment)[https://www.freeipa.org/page/Contribute/Code#create-pull-request-on-github] and think it will better describe why this change should be merged.

> Will that strptime work? I think the value will be like 20240724200619Z

I saw I was duplicating code that was just below so I simply moved the code upwards and kept the format (this code was already merged so yes it will work or we have a bug)

> The time delta is going to display oddly, like: 3:59:59.999997

I expect user to use `.strftime` because they get the raw python object.

`expiration_delta` print like: `1:30:02.143800` but `expiration_delta.days`: `1`
`expiration_datetime`: `2024-07-16 14:00:00+00:00` and in our template `expiration_datetime.strftime("%02d/%02m/%04Y")`: `16/07/2024` (dd/mm/yy like French format)

rcritten commented:
Please update the man page with the new directives. Be sure to mention that they are python objects and not strings.

Et7f3 commented:
I added text to man pages, reviewed by chat gpt (to improve phrasing), added examples with jinja (to highlight they are python objects). I also added the python exact class for referring to the docs.
The rendered manual section:
![image](https://github.com/user-attachments/assets/b8a626d0-0a4d-46d2-a4bc-c080308df885)

rcritten commented:
Can you rebase your patch? Things are looking good.

flo-renaud commented:
Closing this PR as similar functionality has been provided with this commit: https://codeberg.org/freeipa/freeipa/commit/1cde93e84e7dcf4dcd013426093c0bdf72c39bed
